app/Helpers/StockHelper.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Helpers;
+
+use App\Models\Product;
+
+class StockHelper
+{
+    public static function addQuantityToProductByCode(string $productCode, $quantity): void
+    {
+        $product = Product::where('code', $productCode)->first();
+
+        if ($product) {
+            $status = ProductStatusHelper::checkQuantityAndReturnStatus($product->quantity + $quantity);
+            $product->update([
+                'quantity' => $product->quantity + $quantity,
+                'status' => $status
+            ]);
+        }
+    }
+
+    public static function takeQuantityFromProductByCode(string $productCode, $quantity): void
+    {
+        $product = Product::where('code', $productCode)->first();
+
+        if ($product && $product->quantity >= $quantity) {
+            $status = ProductStatusHelper::checkQuantityAndReturnStatus($product->quantity - $quantity);
+            $product->update([
+                'quantity' => $product->quantity - $quantity,
+                'status' => $status
+            ]);
+        }
+    }
+
+    public static function removeAllQuantityToProducts(object $offerOrOrder): void
```

Comment:
Nie jestem pewien czym jest 'offer' w tej zmiennej $offerOrOrder, ale jeśli jest to jakiś obiekt/model to możesz podawać 2 typy w argumentach funkcji, np.
```php
public static function removeAllQuantityToProducts(Offer|Order $offerOrOrder): void
```

Reply:
Czyli w tej sytuacji argumentem może być obiekt typu Offer albo obiekt typu Order